vyper/semantics/analysis/module.py

```diff
@@ -739,32 +693,42 @@ def visit_FunctionDef(self, node):
         self._add_exposed_function(func_t, node)
 
     def visit_Import(self, node):
-        # import x.y[name] as y[alias]
+        self._add_import(node)
 
-        alias = node.alias
+    def visit_ImportFrom(self, node):
+        self._add_import(node)
 
-        if alias is None:
-            alias = node.name
+    def _add_import(self, node: vy_ast.VyperNode) -> None:
+        import_info = node._metadata["import_info"]
+        # similar structure to import analyzer
+        module_info = self._load_import(node, import_info)
 
-        # don't handle things like `import x.y`
-        if "." in alias:
-            msg = "import requires an accompanying `as` statement"
-            suggested_alias = node.name[node.name.rfind(".") :]
-            hint = f"try `import {node.name} as {suggested_alias}`"
-            raise StructureException(msg, node, hint=hint)
+        import_info._typ = module_info
 
-        self._add_import(node, 0, node.name, alias)
+        self.namespace[import_info.alias] = module_info
 
-    def visit_ImportFrom(self, node):
-        # from m.n[module] import x[name] as y[alias]
-        alias = node.alias or node.name
+    def _load_import(self, node: vy_ast.VyperNode, import_info: ImportInfo) -> Any:
```

Comment:
`node` unused

Reply:
wow! nice find!

Reply:
b5e417f51836be4f93957e3f4403357558fde2ec

Reply:
i think this should be caught by a linter

Reply:
yes, i agree -- i was surprised it was not found

Reply:
created an issue to fix the linter: https://github.com/vyperlang/vyper/issues/4280